src/System.Data.SqlClient/src/System/Data/SqlClient/SqlParameter.cs

```diff
@@ -35,12 +37,28 @@ internal StreamDataFeed(Stream source)
 
     internal class TextDataFeed : DataFeed
     {
+        private static UnicodeEncoding s_defaultEncoding;
+
         internal TextReader _source;
 
         internal TextDataFeed(TextReader source)
         {
             _source = source;
         }
+
+        public static UnicodeEncoding DefaultEncoding
+        {
+            get
+            {
+                UnicodeEncoding encoding = s_defaultEncoding;
+                if (encoding is null)
+                {
+                    encoding = new UnicodeEncoding(false, false); ;
+                    Interlocked.CompareExchange(ref s_defaultEncoding, encoding, null);
```

Comment:
If it is important to always return the same instance
```suggestion
                    encoding = Interlocked.CompareExchange(ref s_defaultEncoding, encoding, null) ?? encoding;
```

Reply:
It isn't massively important but i may as well, done and removed that extra ; on the live above as well.